cmd/kubeadm/app/preflight/checks.go

```diff
@@ -351,6 +353,24 @@ func (sysver SystemVerificationCheck) Check() (warnings, errors []error) {
 	return warns, nil
 }
 
+type KubernetesVersionCheck struct {
+	KubernetesVersion string
+}
+
+func (kubever KubernetesVersionCheck) Check() (warnings, errors []error) {
+	k8sVersion, err := version.ParseSemantic(kubever.KubernetesVersion)
+	if err != nil {
+		return nil, []error{fmt.Errorf("couldn't parse kubernetes version %q: %v", kubever.KubernetesVersion, err)}
+	}
+
+	kubeadmVersion := apiversion.Get()
+	if compare, _ := k8sVersion.Compare(kubeadmVersion.GitVersion); compare == 1 {
```

Comment:
Agreed -- my original intent in the issue was for our 1.X releases, not patch releases.

As for the comparison itself, rather than having to cognitively remember what a positive vs. negative return value mean, this might read cleaner using `.LessThan` / `.AtLeast` instead so it's clearer which side you expect to be larger than the other. However, given @luxas's comment, you'll probably drop the semver comparison altogether.